### PR TITLE
mediasynclite: init at 0.4.2

### PR DESCRIPTION
**Pull request by tobz619**
## Description of changes
Packaged iBroadcast Mediasynclite. Homepage is found [here](https://github.com/iBroadcastMediaServices/MediaSyncLiteLinux). I have learned how to fetch from upstream and use my edits.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ pkgs
+, lib
+}:
+
+pkgs.stdenv.mkDerivation rec {
+  pname = "mediasynclite";
+  version = "0.4.2";
+
+  src = pkgs.fetchgit {
```
Comment (sikmir) [inline pkgs/by-name/me/mediasynclite/package.nix:9]: Use `fetchFromGitHub`

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ pkgs
```
Comment (sikmir) [inline pkgs/by-name/me/mediasynclite/package.nix:1]: Don't use `pkgs`

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   curl
+   openssl.dev
+   jansson
+   wrapGAppsHook
+  ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  prePatch = ''
+    substituteAllInPlace ./src/ibmsl.c
```
Comment (sikmir) [inline pkgs/by-name/me/mediasynclite/package.nix:39]: What is replaced here?

Comment (tobz619) [reply pkgs/by-name/me/mediasynclite/package.nix:39]: > What is replaced here?

The @out@ [here](https://github.com/tobz619/MediaSyncLiteLinuxNix/blob/d0d5caa2a1bb260f8081e4cf49cb0219b123f682/src/ibmsl.c#L671)

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mediasynclite";
+  version = "0.4.2";
+
+  src = fetchFromGitHub {
+    owner = "iBroadcastMediaServices";
+    repo = "MediaSyncLiteLinux";
+    rev = "d0d5caa2a1bb260f8081e4cf49cb0219b123f682";
```
[the remark this answers is not in the document]
Comment (tobz619) [reply pkgs/by-name/me/mediasynclite/package.nix:21]: > 

Hi, I don't understand this change. From what I'm aware of, I though the rev was supposed to be the git commit hash of the changes I want to make to the upstream branch?

[the remark this answers is not in the document]
Comment (sikmir) [reply pkgs/by-name/me/mediasynclite/package.nix:21]: No, git hash is used only if no proper git tag in upstream repo.

File: pkgs/by-name/me/mediasynclite/package.nix (hunk 3 above)
Comment (sikmir) [reply pkgs/by-name/me/mediasynclite/package.nix:39]: But in such case you need `--subst-var out`

**Comment by sikmir** [comment]
```
       > clang -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-align -Wwrite-strings -Wredundant-decls -Wformat-nonliteral -Wnested-externs -pedantic -std=gnu99 -O3 -pthread -D_REENTERANT `pkg-config --cflags gtk+-3.0 libcurl libssl libcrypto` -c ../src/tools.c -o tools.o
       > ../src/tools.c:490:16: error: call to undeclared library function 'strcasecmp' with type 'int (const char *, const char *)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       >             if(strcasecmp(ext, profile.supported_exts[i]) != 0)
       >                ^
       > ../src/tools.c:490:16: note: include the header <strings.h> or explicitly provide a declaration for 'strcasecmp'
       > 1 error generated.
       > make[1]: *** [Makefile:38: tools.o] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-mediasynclite-0.4.2.drv-0/source/src'
       > make: *** [Makefile:12: all] Error 2
```

**Comment by tobz619** [comment]
> ```
>        > clang -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-align -Wwrite-strings -Wredundant-decls -Wformat-nonliteral -Wnested-externs -pedantic -std=gnu99 -O3 -pthread -D_REENTERANT `pkg-config --cflags gtk+-3.0 libcurl libssl libcrypto` -c ../src/tools.c -o tools.o
>        > ../src/tools.c:490:16: error: call to undeclared library function 'strcasecmp' with type 'int (const char *, const char *)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
>        >             if(strcasecmp(ext, profile.supported_exts[i]) != 0)
>        >                ^
>        > ../src/tools.c:490:16: note: include the header <strings.h> or explicitly provide a declaration for 'strcasecmp'
>        > 1 error generated.
>        > make[1]: *** [Makefile:38: tools.o] Error 1
>        > make[1]: Leaving directory '/private/tmp/nix-build-mediasynclite-0.4.2.drv-0/source/src'
>        > make: *** [Makefile:12: all] Error 2
> ```

I haven't been able to replicate this. How did you do this?

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    owner = "iBroadcastMediaServices";
+    repo = "MediaSyncLiteLinux";
+    rev = version;
+    hash = "sha256-mS2e6zZutSEAr5gCS58t0Nzg8dCJgn/yoFMmbgnIM2I=";
+    };
+
+  buildInputs = [
+   gtk3
+   glib
+   gsettings-desktop-schemas
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:28]: This probably needs to go to nativeBuildInputs, too.

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+   curl
+   openssl.dev
+   jansson
+  ];
+
+  nativeBuildInputs = [
+   pkg-config
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:35]: Please use 2 spaces to intend

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+   wrapGAppsHook
+  ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  prePatch = ''
+    substitute ./src/ibmsl.c ./src/ibmsl.c --subst-var out
+    '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:43]:
```suggestion
  postPatch = ''
    substitute ./src/ibmsl.c ./src/ibmsl.c --subst-var out
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please clean up the commit history

File: pkgs/by-name/me/mediasynclite/package.nix (hunk 5 above)
Comment (tobz619) [reply pkgs/by-name/me/mediasynclite/package.nix:28]: Is there a source that I can find to help me figure out what goes in `nativeBuildInputs` vs `buildInputs`?

**Comment by SuperSandro2000** [comment]
`mediasynclite: init at 0.4.2` and `Final package.nix` should be squashed, too.

**Comment by tobz619** [comment]
I believe this is ready for review

**Review by tobz619 — COMMENTED** [review]
Commit history cleaned

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+stdenv.mkDerivation rec {
+  pname = "mediasynclite";
+  version = "0.4.2";
+
+  src = fetchFromGitHub {
+    owner = "iBroadcastMediaServices";
+    repo = "MediaSyncLiteLinux";
+    rev = version;
+    hash = "sha256-mS2e6zZutSEAr5gCS58t0Nzg8dCJgn/yoFMmbgnIM2I=";
+    };
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:23]:
```suggestion
  };
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+   openssl.dev
+   jansson
+  ];
+
+  nativeBuildInputs = [
+   gtk3
+   glib
+   gsettings-desktop-schemas
+   pkg-config
+   wrapGAppsHook
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:36]: Please rebase and replace this with wrapGAppsHook3

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+  buildInputs = [
+   curl
+   openssl.dev
+   jansson
+  ];
+
+  nativeBuildInputs = [
+   gtk3
+   glib
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:33]: I would inuitively move them to buildInputs rather but we can clarify this by setting strictDeps to true

Comment (tobz619) [reply pkgs/by-name/me/mediasynclite/package.nix:35]: I've added strictDeps: [earlier](https://github.com/NixOS/nixpkgs/pull/294716#discussion_r1554569226) you asked me to move them to `nativeBuildInputs`

Comment (SuperSandro2000) [reply pkgs/by-name/me/mediasynclite/package.nix:35]: That comment was only about line 28, gsettings-desktop-schemas, not the entire list.

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+   curl
+   openssl.dev
+   jansson
+  ];
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+   gtk3
+   glib
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:35]:
```suggestion
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+  src = fetchFromGitHub {
+    owner = "iBroadcastMediaServices";
+    repo = "MediaSyncLiteLinux";
+    rev = version;
+    hash = "sha256-mS2e6zZutSEAr5gCS58t0Nzg8dCJgn/yoFMmbgnIM2I=";
+    };
+
+  buildInputs = [
+   curl
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:26]:
```suggestion
   curl
   glib
   gtk3
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+   openssl.dev
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:27]:
```suggestion
   openssl
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+   jansson
+  ];
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+   gsettings-desktop-schemas
+   pkg-config
+   wrapGAppsHook3
+  ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:39]:
```suggestion
  nativeBuildInputs = [
    gsettings-desktop-schemas
    pkg-config
    wrapGAppsHook3
  ];
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    hash = "sha256-mS2e6zZutSEAr5gCS58t0Nzg8dCJgn/yoFMmbgnIM2I=";
+    };
+
+  buildInputs = [
+   curl
+   glib
+   gtk3
+   openssl
+   jansson
+  ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:31]:
```suggestion
  buildInputs = [
    curl
    glib
    gtk3
    openssl
    jansson
  ];
```

File: pkgs/by-name/me/mediasynclite/package.nix [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+stdenv.mkDerivation rec {
+  pname = "mediasynclite";
+  version = "0.4.2";
+
+  src = fetchFromGitHub {
+    owner = "iBroadcastMediaServices";
+    repo = "MediaSyncLiteLinux";
+    rev = version;
+    hash = "sha256-mS2e6zZutSEAr5gCS58t0Nzg8dCJgn/yoFMmbgnIM2I=";
+    };
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/mediasynclite/package.nix:23]:
```suggestion
  };
```

**Comment by SuperSandro2000** [comment]
In the future please do not create PRs from the master branch, as maintainers cannot force push into that branch to quickly fix last minute issues.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by tobz619** [comment]
Woo! Thank you :)

**Pull request merged**